compiler/src/dotty/tools/dotc/core/Constraint.scala

```diff
@@ -172,6 +174,23 @@ abstract class Constraint extends Showable {
    */
   def occursAtToplevel(param: TypeParamRef, tp: Type)(using Context): Boolean
 
+  /** Sanitize `bound` to make it either a valid upper or lower bound for
+   *  `param` depending on `isUpper`.
+   *
+   *  Toplevel references to `param`, are replaced by `Any` if `isUpper` is true
```

Comment:
What's a toplevel reference?

Reply:
It's explained in the `@see` just below :)

Reply:
Ah, found it.

```scala
  /** Does `param` occur at the toplevel in `tp` ?
   *  Toplevel means: the type itself or a factor in some
   *  combination of `&` or `|` types.
   */
  def occursAtToplevel(param: TypeParamRef, tp: Type)(using Context): Boolean
```